src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -229,6 +229,19 @@ static term create_pair(Context *ctx, term term1, term term2)
     return ret;
 }
 
+static term reason_atom_to_error_tuple_safely(Context *ctx, term err_atom)
+{
+    if (UNLIKELY(!term_is_atom(err_atom))) {
```

Comment:
This is a static function, so it means we have complete control and knowledge about how it is used.
Do we know any situation where this happen?

Reply:
Not currently, I added this, as suggested by @fadushin in the top "resolved" conversation, as a safeguard for future maintainers making changes to the driver.

>You might want to add a check to ensure that err-atom is always an atom, or at least document so, so that future maintainers don't pass in a tuple expecting things to work.